src/network/networkd-dhcp6.c

```diff
@@ -1456,27 +1446,37 @@ static bool dhcp6_pd_uplink_is_ready(Link *link) {
 }
 
 int dhcp6_pd_find_uplink(Link *link, Link **ret) {
-        Link *l;
+        Link *uplink = NULL;
+        int r = 0;
 
         assert(link);
         assert(link->manager);
-        assert(link->network);
+        assert(link_dhcp6_pd_is_enabled(link));
         assert(ret);
 
-        if (dhcp6_pd_resolve_uplink(link, &l) >= 0) {
-                if (!dhcp6_pd_uplink_is_ready(l))
+        if (link->network->dhcp6_pd_uplink_name)
```

Comment:
`assert(link->network)` or check

Reply:
Also here. That is already checked by `link_dhcp6_pd_is_enabled()` in the above.